### ISSUE TITLE
VDesk hangs after command runs in console window

### ISSUE THREAD

**Issue by admattingly** [LittleVaaty/VDesk#6]
If I use VDesk to "run" a console command, the C:\Windows\System32\cmd.exe window does not close once the command completes (successful) execution -it just sits there until I manually close the window.  This is happening on Windows 11.

**Comment by LittleVaaty** [comment]
If you do `vdesk run -o 1  C:\Windows\System32\cmd.exe` it's the expected behavior that after the command is succesfully finshed, the cmd window is still open.
Maybe `vdesk open` will be a better command name.

**Comment by admattingly** [comment]
I strongly disagree.  The only reason cmd.exe hangs around is because this is the default behaviour of cmd.exe.  If you run cmd.exe with the /C option, it terminates after running whatever command you specify after the /C.

 

I think the behaviour of vdesk should be to open a console application on the specified desktop and remove its console window once the executable terminates.

 

If this is not acceptable as the default behaviour of vdesk, perhaps an option could be added to invoke this behaviour?

 

Kind regards

Andrew.

**Comment by LittleVaaty** [comment]
If i understand correctly, you wan that after any run command, vdesk closes the shell in which it is executed ?

**Comment by admattingly** [comment]
Yes, once the command/program has completely finished executing and has returned control to its parent (shell), the console window should be closed.

**Comment by LittleVaaty** [comment]
Then what appen when vdesk is executed in a script ?
vdesk return the code 0 when success after that it is up to the parent terminal  or script to act accordinly.
vdesk has no control over the parent process

**Comment by DoS007** [comment]
Just to add context: I am pretty sure that the original VDesk didn't block that way. But this is just a design decision. Currently appropriate usage is probably using `start` like in the sample.bat of the project if you want a similar behaviour. I use that as well.

**Comment by admattingly** [comment]
Just to add my context:  I’m effectively running a script where each step needs to complete before initiating the next step.  Each step involves running a console program which pops up in its own window, for seconds/minutes.  I can’t use start, as this returns to the script immediately.  I was hoping to use VDesk to make the console windows pop up in another desktop, so they are less of a nuisance.  I can’t control these console windows to force them to run minimised.

 

Kind regards

 

Andrew Mattingly

M: +61 411 263 441

E:  ***@***.***> ***@***.***

R: VK2GCO

**Comment by DoS007** [comment]
You could use for hiding the `/MIN` Option of `start`. You can also use the Task Scheduler for autostarting and then choose the option hidden. This might even hide the plop in the status/task bar.

**Comment by LittleVaaty** [comment]
vdesk will just start the scipt but will not wait the end of execution of the script.
I'm not familiar with batch, but if powershell is an option, Start-Process has an 'wait' option that will do what you want.

**Comment by admattingly** [comment]
I can’t use start, as I said, because it returns control immediately and I need my calling script to suspend until the console program completes.Sent from my iPhoneOn 3 May 2024, at 1:02 AM, Jonathan Hansen ***@***.***> wrote:﻿
vdesk will just start the scipt but will not wait the end of execution of the script.
I'm not familiar with batch, but if powershell is an option, Start-Process has an 'wait' option that will do what you want.

**Comment by DoS007** [comment]
@admattingly You probably didn't getmy Edit-Update via E-Mail. But for your case, you should really try the Windows Task Scheduler with the hidden Option.
